File: python/001.py
##############################################################
# Q: Find the sum of all the multiples of 3 or 5 below 1000. #
##############################################################

import functools


def get_sum(arr):
    return functools.reduce(lambda x, y: x + y, arr)


def is_multiple(number):
    if number % 3 == 0 or number % 5 == 0:
        return True
    else:
        return False


def gen_multiples(max):
    multiples = []
    for x in range(1, max):
        if is_multiple(x):
            multiples.append(x)

    return multiples


def get_mult_sum(number):
    multiples = gen_multiples(number)
    sum = get_sum(multiples)
    return sum


sum = get_mult_sum(1000)
print(sum)  # 233168

File: go/001.go
/*
Q: Find the sum of all the multiples of 3 or 5 below 1000.
*/

package main

func getSum(numbers []int) int {
	sum := 0

	for _, number := range numbers {
		sum += number
	}
	return sum
}

func isMultiple(number int) bool {
	return number%3 == 0 || number%5 == 0 || false
}

func genMultiples(max int) []int {
	multiples := []int{}
	for i := 1; i < max; i++ {
		if isMultiple(i) {
			multiples = append(multiples, i)
		}
	}
	return multiples
}

func getMultSum(max int) int {
	multiples := genMultiples(max)
	sum := getSum(multiples)
	return sum
}

func main() {
	println(getMultSum(1000)) // 233168
}
